apps/sms/services/js/messaging/messaging_server.js

```diff
@@ -0,0 +1,542 @@
+    // isRead == false i.e mark as unread case, marking only one message
+    // as unread is sufficient.
+    req.onsuccess = (function onsuccess() {
+      if (!list.length || !isRead) {
+        return;
+      } else if (isRead) {
+        this.markMessagesRead(list, isRead);
+      }
+    }).bind(this);
+
+    req.onerror = function onerror() {
+      console.error(
+        'Error while marking message %d as read: %s', id, this.error.name
+      );
+    };
+  },
+
+  getSegmentInfo: function ms_getSegmentInfo(text) {
+    if (!(mozMobileMessage && mozMobileMessage.getSegmentInfoForText)) {
+      return Promise.reject(new Error('mozMobileMessage is unavailable.'));
+    }
+
+    var defer = Utils.Promise.defer();
+
+    var request = mozMobileMessage.getSegmentInfoForText(text);
+    request.onsuccess = function onsuccess(e) {
+      defer.resolve(e.target.result);
+    };
+
+    request.onerror = function onerror(e) {
+      defer.reject(e.target.error);
+    };
+
+    return defer.promise;
+  },
+
+  /**
+   * findThreadFromNumber
+   *
+   * Find a SMS/MMS thread from a number.
+   * @return Promise that resolve to a threadId or rejected if not found
+   */
+  findThreadFromNumber: function ms_findThread(number) {
+
+    function checkCandidate(message) {
+      var isMessageInThread =
+        isMessageBelongTo1to1Conversation(number, message);
+
+      if (isMessageInThread) {
+        threadId = message.threadId;
+        // we need to set the current threadId,
+        // because we start sms app in a new window
+        Threads.registerMessage(message);
+        return false; // found the message, stop iterating
+      }
+    }
+
+    var threadId = null;
+    var deferred = Utils.Promise.defer();
+
+    MessagingService.getMessages({
+      filter: { numbers: [number] },
+      each: checkCandidate,
+      done: function() {
+        if (threadId == null) {
+          deferred.reject(new Error('No thread found for number: ' + number));
+          return;
+        } else {
+          deferred.resolve(threadId);
+          return;
+        }
+      }
+    });
+
+    return deferred.promise;
+  }
+};
+
+MessagingService.initEventListener();
+
+bridge.service('messaging')
+  .method('getMessage', MessagingService.getMessage)
+  .method('retrieveMMS', MessagingService.retrieveMMS)
+  .method('sendSMS', MessagingService.sendSMS)
+  .method('sendMMS', MessagingService.sendMMS)
+  .method('resendMessage', MessagingService.resendMessage)
+  .method('deleteMessages', MessagingService.deleteMessages)
+  .method('markThreadRead', MessagingService.markThreadRead)
+  .method('markMessageRead', MessagingService.markMessageRead)
+  .method('getSegmentInfo', MessagingService.getSegmentInfo)
+  .method('findThreadFromNumber', MessagingService.findThreadFromNumber)
+  .stream('getThreadStream', MessagingService.getThreadStream)
+  .stream('getMessageStream', MessagingService.getMessageStream);
```

Comment:
We can keep it like this for now but eventually we should try to have less methods.. but not sure yet how this could look like.

The main focus right now is to split services and make the app work so I'm fine if we don't refactor this much more.